apiserver/params/storage.go

```diff
@@ -33,3 +33,28 @@ type StorageShowResult struct {
 type StorageListResult struct {
 	Instances []StorageInstance
 }
+
+// PoolInstance holds data for a pool instance.
+type PoolInstance struct {
+	Name   string
+	Type   string
+	Traits map[string]interface{}
+}
+
+// PoolListFilter holds a filter for pool list API call.
+type PoolListFilter struct {
+	Names []string
+	Types []string
+}
+
+// PoolListResults holds a collection of pool instances.
+type PoolListResults struct {
```

Comment:
I don't think these types make sense. It's not really a bulk call, it's a single call with a single set of filter attributes. I think this ought to be:

```
type StoragePoolsResult struct {
    Pools []StoragePool
}
```